### PR TITLE
Initial translation of hooks-rules

### PR DESCRIPTION
**Pull request by shakib609**

### CODE REVIEW

File: content/docs/hooks-rules.md [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
 id: hooks-rules
-title: Rules of Hooks
+title: Hook ব্যবহারের নিয়মাবলী
 permalink: docs/hooks-rules.html
 next: hooks-custom.html
 prev: hooks-effect.html
 ---
 
-*Hooks* are a new addition in React 16.8. They let you use state and other React features without writing a class.
+*Hook* হল React 16.8 এর নতুন সংযোজন। এটি আপনাকে ক্লাস ব্যবহার না করেই state এবং React এর অন্যান্য ফিচার ব্যবহারের সুযোগ করে দেয়।
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: Missing `s` from Hook**s**

File: content/docs/hooks-rules.md [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
 
-Hooks are JavaScript functions, but you need to follow two rules when using them. We provide a [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) to enforce these rules automatically:
+Hook হল জাভাস্ক্রিপ্ট ফাংশন, কিন্তু আপনাকে এটি ব্যবহারের সময় দুইটি নিয়ম মেনে চলতে হবে। আমরা একটি [linter plugin](https://www.npmjs.com/package/eslint-plugin-react-hooks) প্রদান করে থাকি যাতে স্বয়ংক্রিয়ভাবে এই নিয়মগুলো কার্যকর হয়ঃ
 
-### Only Call Hooks at the Top Level {#only-call-hooks-at-the-top-level}
+### শুধুমাত্র টপ লেভেলে Hook কল করুন {#only-call-hooks-at-the-top-level}
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: Hooks

File: content/docs/hooks-rules.md [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
 
-**Don't call Hooks inside loops, conditions, or nested functions.** Instead, always use Hooks at the top level of your React function. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That's what allows React to correctly preserve the state of Hooks between multiple `useState` and `useEffect` calls. (If you're curious, we'll explain this in depth [below](#explanation).)
+**কোন লুপ, শর্ত বা nested ফাংশনের ভেতর Hooks কল করবেননা।** এর পরিবর্তে, সবসময় আপনার React ফাংশনের টপ লেভেলে Hook কল করুন। এই নিয়ম মেনে চলার মাধ্যমে আপনি নিশ্চিত করেন যে, প্রতিবার রি-রেন্ডারের সময় Hook গুলো একই ক্রমানুসারে কল করা হবে। এটি অনেকগুলো `useState` এবং `useEffect` কলের মাঝে React কে সঠিকভাবে Hook গুলোর স্টেট সংরক্ষণ করতে সাহায্য করে। (আপনি যদি কৌতূহলী হয়ে থাকেন, আমরা [নিচে](#explanation) এটি সম্পর্কে আরও গভীর আলোচনা করব।)
 
-### Only Call Hooks from React Functions {#only-call-hooks-from-react-functions}
+### শুধুমাত্র React Function এর ভেতর Hook কল করুন {#only-call-hooks-from-react-functions}
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: Let's consistent with Function. We can translate  it as ফাংশন

File: content/docs/hooks-rules.md [hunk 4]
```diff
@@ -46,13 +46,13 @@ npm install eslint-plugin-react-hooks --save-dev
 }
 ```
 
-In the future, we intend to include this plugin by default into Create React App and similar toolkits.
+আমরা ভবিষ্যতে এই plugin টি শুরু থেকেই Create React App এবং একই ধরণের টুলকিটগুলোর সাথে সংযুক্ত করে দেয়ার ইচ্ছা আছে।
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: <del>আমরা</del> -> আমাদের

File: content/docs/hooks-rules.md [hunk 5]
```diff
@@ -118,9 +118,9 @@ useState('Poppins')        // 🔴 2 (but was 3). Fail to read the surname state
 useEffect(updateTitle)     // 🔴 3 (but was 4). Fail to replace the effect
 ```
 
-React wouldn't know what to return for the second `useState` Hook call. React expected that the second Hook call in this component corresponds to the `persistForm` effect, just like during the previous render, but it doesn't anymore. From that point, every next Hook call after the one we skipped would also shift by one, leading to bugs.
+React দ্বিতীয় `useState` Hook কলের জন্য কি রিটার্ন করবে বুঝতে পারবেনা। React আশা করছিল এই কম্পোনেন্টের  দ্বিতীয় Hook কল যাতে `persistForm` effect এর সাথে সঙ্গতিপূর্ণ হয়, ঠিক এর পূর্ববর্তি রেন্ডারের মত, কিন্তু এখন এরা আর সঙ্গতিপূর্ণ নয়। এর পর থেকে, আমাদের বাদ দেয়া প্রতিটি Hook কলও এক ধাপ সরে যাবে, যা পরবর্তিতে বাগের সৃষ্টি করবে।
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: বাগের -> bug ?

File: content/docs/hooks-rules.md [hunk 6]
```diff
@@ -118,9 +118,9 @@ useState('Poppins')        // 🔴 2 (but was 3). Fail to read the surname state
...
 
-**This is why Hooks must be called on the top level of our components.** If we want to run an effect conditionally, we can put that condition *inside* our Hook:
+**এই জন্যই Hook গুলো সবসময় আমাদের কম্পোনেন্টের সর্বোচ্চ স্তরে কল করতে হবে।** আমরা যদি কোন effect শর্তসাপেক্ষে রান করাতে চাই, আমরা ঐ শর্ত আমাদের Hook *ভেতরে* রাখতে পারিঃ
```
Comment (nutboltu) [inline content/docs/hooks-rules.md]: Hook এর *ভেতরে* রাখতে পারিঃ

**Comment by shakib609** [comment]
Updated the translation according to your review. Can you please take another look?

**Pull request merged**